### PR TITLE
Convert dtls_mtu_test, dtlsv1listentest

### PR DESCRIPTION
**Pull request by richsalz**
Also converted most of ssltestlib but left the packet_dump output
as-is (for now?).

### CODE REVIEW

File: test/build.info [hunk 1]
```diff
@@ -219,7 +219,7 @@ IF[{- !$disabled{tests} -}]
   INCLUDE[lhash_test]=.. ../include
   DEPEND[lhash_test]=../libcrypto
 
-  SOURCE[dtlsv1listentest]=dtlsv1listentest.c
+  SOURCE[dtlsv1listentest]=dtlsv1listentest.c testutil.c test_main.c
```
Comment (levitte) [inline test/build.info]: Remove this change

File: test/build.info [hunk 2]
```diff
@@ -219,7 +219,7 @@ IF[{- !$disabled{tests} -}]
...
   INCLUDE[dtlsv1listentest]=.. ../include
   DEPEND[dtlsv1listentest]=../libssl
```
Comment (levitte) [inline test/build.info]: Add ` libtestutil.a` here

File: test/build.info [hunk 3]
```diff
@@ -298,7 +298,7 @@ IF[{- !$disabled{tests} -}]
 
   IF[{- !$disabled{psk} -}]
     PROGRAMS_NO_INST=dtls_mtu_test
-    SOURCE[dtls_mtu_test]=dtls_mtu_test.c ssltestlib.c
+    SOURCE[dtls_mtu_test]=dtls_mtu_test.c ssltestlib.c testutil.c test_main.c
```
Comment (levitte) [inline test/build.info]: Remove this change

File: test/build.info [hunk 4]
```diff
@@ -298,7 +298,7 @@ IF[{- !$disabled{tests} -}]
...
     INCLUDE[dtls_mtu_test]=.. ../include
     DEPEND[dtls_mtu_test]=../libcrypto ../libssl
```
Comment (levitte) [inline test/build.info]: Add ` libtestutil.a` here

File: test/dtls_mtu_test.c [hunk 5]
```diff
@@ -15,6 +15,8 @@
 #include <openssl/err.h>
 
 #include "ssltestlib.h"
+#include "testutil.h"
+#include "test_main.h"
```
Comment (levitte) [inline test/dtls_mtu_test.c]: Remove this line

File: test/dtlsv1listentest.c [hunk 6]
```diff
@@ -12,10 +12,9 @@
 #include <openssl/bio.h>
 #include <openssl/err.h>
 #include <openssl/conf.h>
-#ifndef OPENSSL_NO_ENGINE
- #include <openssl/engine.h>
-#endif
 #include "e_os.h"
+#include "testutil.h"
+#include "test_main.h"
```
Comment (levitte) [inline test/dtlsv1listentest.c]: Remove this line

**Comment by richsalz** [comment]
addressed all issues, rebased, updated commit pushed.

**Comment by richsalz** [comment]
there's a travis failure, but not related to this: https://travis-ci.org/openssl/openssl/jobs/225605474
and the appveyor failure also seems not related: https://ci.appveyor.com/project/openssl/openssl/build/master.9863/job/6qm4e0bl4ijamn9j#L1956

**Comment by richsalz** [comment]
Just in case, rebase and force-pushing.

**Comment by mattcaswell** [comment]
Travis failure looks unrelated. Can this be merged?

**Pull request closed without merging**